odl/discr/grid.py

```diff
@@ -246,72 +223,81 @@ def max_pt(self):
-        `as_midp` ``True``, in which case the minimum is half a cell
-        smaller than the minimum grid point.
+        See also
+        --------
+        Intervalprod.min
 
         Examples
         --------
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=False)
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
         >>> g.min()
         array([ 1., -2.])
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=True)
-        >>> g.min()
-        array([ 0.5 , -3.75])
         """
-        if not self.as_midp:
-            return self.min_pt
-        elif self._exact_min is not None:
-            return self._exact_min
-        else:
-            minpt_cell_size = np.array([cs[0] for cs in self.cell_sizes()])
-            return self.min_pt - minpt_cell_size / 2
+        return self.min_pt
 
     def max(self):
-        """Return vector with maximal cell coordinates per axis.
+        """Return `max_pt`.
 
-        This is relevant if the grid was initialized with
-        `as_midp` ``True``, in which case the maximum is half a cell
-        larger than the maximum grid point.
+        See also
+        --------
+        Intervalprod.max
 
         Examples
         --------
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=False)
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
         >>> g.max()
         array([ 5.,  2.])
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=True)
-        >>> g.max()
-        array([ 6.5 ,  2.25])
         """
-        if not self.as_midp:
-            return self.max_pt
-        elif self._exact_max is not None:
-            return self._exact_max
-        else:
-            maxpt_cell_size = np.array([cs[-1] for cs in self.cell_sizes()])
-            return self.max_pt + maxpt_cell_size / 2
-
-    # Methods
+        return self.max_pt
 
     def extent(self):
-        """Return a vector containing the total grid extent."""
-        return self.max() - self.min()
+        """Return a vector containing the total grid extent.
+
+        Examples
+        --------
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
+        >>> g.extent()
+        array([ 4.,  4.])
+        """
+        return self.max_pt - self.min_pt
+
+    def convex_hull(self):
+        """The smallest `IntervalProd` containing this grid.
+
+        The convex hull of a set is the union of all line segments
+        between points in the set. For a tensor grid, it is the
+        interval product given by the extremal coordinates.
+
+        Returns
+        -------
+        chull : `IntervalProd`
+            Interval product defined by the minimum and maximum of
+            the grid
+
+        Examples
+        --------
+        >>> g = TensorGrid([-1, 0, 3], [2, 4], [5], [2, 4, 7])
+        >>> g.convex_hull()
+        IntervalProd([-1.0, 2.0, 5.0, 2.0], [3.0, 4.0, 5.0, 7.0])
+        """
+        return IntervalProd(self.min(), self.max())
 
     def element(self):
```

Comment:
Perhaps related to the missing upstream squeeze fix. I'll check what it looks like after the next rebase.